### PR TITLE
M0078DDG-316 add javadoc references to platform developer guide

### PR DESCRIPTION
**Pull request by fpinel-microej**

### CODE REVIEW

**Comment by fpinel-microej** [comment]
ecom-comm : l.276&277 `getPlatformId()` and `getName()` methods are mentionned but not in Javadoc, are these maybe not in Java ?

**Comment by fpinel-microej** [comment]
uiImageGenerator : There are a bunch of methods mentioned from lines 162 to 184 that aren't in the Javadoc, is this normal ?

**Comment by fpinel-microej** [comment]
uiChangelog : l.883 there doesn't seem to be a `Image.createImage(stream)` in Javadoc, has it been removed maybe ?

**Comment by fpinel-microej** [comment]
uiMigrationGuide : Method `initialize()` is mentioned l.158, which class is it from ? 
`flush(MicroUIGraphicsContext, Image, int, int, int, int)` is also mentioned l.159, but there doesn't seem to a `flush` with this signature in the Javadoc.

**Comment by fpinel-microej** [comment]
frontpanel.rst : There are a bunch of methods supposedly from Widget that are mentioned around lines 230-240, but they are not in the `ej.mwt.Widget` class in the Javadoc. Is it another Widget class ?

File: PlatformDeveloperGuide/architectureChangelog.rst [hunk 1]
```diff
@@ -69,17 +71,21 @@ Simulator
 ~~~~~~~~~
 
 -  Added class file major version check (<=51). Classes must be compiled for Java 7 or lower. Set the options property ``S3.DisableClassFileVersionCheck`` to ``false`` to disable this verification.
--  Added native method signature in the stack trace of the `UnsatisfiedLinkError <https://repository.microej.com/javadoc/microej_5.x/apis/java/lang/UnsatisfiedLinkError.html>`_ thrown when a native method is missing.
+-  Added native method signature in the stack trace of the `UnsatisfiedLinkError`_ thrown when a native method is missing.
 -  Fixed HIL engine method ``NativeInterface.getResourceContent()`` that generates a runtime error in the Simulator.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/architectureChangelog.rst:75]: Yes, HIL is a tool so it does not appear in the central repository javadoc.

File: PlatformDeveloperGuide/uiChangeLog.rst [hunk 2]
```diff
@@ -521,9 +539,11 @@ MicroUI Implementation
 **Fixed**
 
 * Pending flush cannot be added after an ``OutOfEventException``.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/uiChangeLog.rst:541]: Internal class. Let it as is.

File: PlatformDeveloperGuide/architectureChangelog.rst [hunk 3]
```diff
@@ -863,28 +913,33 @@ Simulator
...
    (mutable) object instead of an immutable one
--  Fixed `Object.clone() <https://repository.microej.com/javadoc/microej_5.x/apis/java/lang/Object.html#clone-->`_ crash when an OutOfMemory occurs
+-  Fixed `Object.clone()`_ crash when an OutOfMemory occurs
 -  Fixed potential crash when calling an abstract method (some
    interfaces of the hierarchy were not taken into account - introduced
    in version :ref:`7.10.0 <changelog-7.10.0>`)
 -  Fixed ``OutOfMemory`` errors even if the heap is not full (resources
-   loaded from `Class.getResourceAsStream() <https://repository.microej.com/javadoc/microej_5.x/apis/java/lang/Class.html#getResourceAsStream-java.lang.String->`_
+   loaded from `Class.getResourceAsStream()`_
    and ``ResourceBuffer.open()`` were taken into account in simulated heap
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/architectureChangelog.rst:929]: replaced by type

File: PlatformDeveloperGuide/ecom.rst [hunk 4]
```diff
@@ -41,41 +41,47 @@ connection on a hardware device.
    connection kind and its configuration are fixed by the parameter
    ``String connectionString`` of the method ``Connection.open``.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/ecom.rst:42]: done (Connector)

File: PlatformDeveloperGuide/uiImageGenerator.rst [hunk 5]
```diff
@@ -297,14 +299,15 @@ Use
 ===
 
 The MicroUI Image APIs are available in the class
-`ej.microui.display.Image <https://repository.microej.com/javadoc/microej_5.x/apis/ej/microui/display/Image.html#>`_ ant its subclasses. There are no specific APIs that use a
+`ej.microui.display.Image`_ ant its subclasses. There are no specific APIs that use a
 pre-generated image. When an image has been pre-processed, the MicroUI
 Image APIs ``getImage`` and ``loadImage`` will get/load the images.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/uiImageGenerator.rst:304]: It is in ResourceImage. Let it as is.

File: PlatformDeveloperGuide/uiImageLoader.rst [hunk 6]
```diff
@@ -86,6 +88,7 @@ This chapter describes the steps to open an external resource from the applicati
 
 .. note:: The simulator (Front Panel) does not manage the external resources. All images listed in ``.imagesext.list`` files are generated in the external resources folder, and this folder is added to the simulator's classpath. 
 
+.. _Image.loadImage(String): https://repository.microej.com/javadoc/microej_5.x/apis/ej/microui/display/ResourceImage.html#loadImage-java.lang.String-
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/uiImageLoader.rst:91]: done

**Comment by frivieremicroej** [comment]
- frontpanel.rst : here we speak about Front Panel Widgets (tool library that does not appear in javadoc)
- uiMigrationGuide : here we speak about LLAPIs (not relevant)
- uiChangelog: DONE
- ecom-comm & uiImageGenerator here we speak about a tool (not relevant)

File: PlatformDeveloperGuide/uiChangeLog.rst [hunk 7]
```diff
@@ -956,12 +980,14 @@ MicroUI Implementation
 
 * ImageScale bugs.
 * Draw string: some errors are not thrown.
-* ``Font.getWidth()`` and ``getHeight()`` don't use ratio factor.
+* ``Font.getWidth()`` and `getHeight()`_ don't use ratio factor.
```
[the remark this answers is not in the document]
Comment (gbalan-microej) [reply PlatformDeveloperGuide/uiChangeLog.rst:983]: wrong note. API (UI2) stringWidth, charWidth, substringWidth.
note: the links **must not** target MicroUI3 API!

[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/uiChangeLog.rst:983]: OK let it as is

**Review by frivieremicroej — APPROVED** [review]

**Pull request merged**